File: code_challenge_solution/java/177 Simon.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;
import java.util.Scanner;

class Game {
    private List<Integer> sequence;
    private int level;
    private int currentIndex;

    public Game() {
        sequence = new ArrayList<>();
        level = 1;
        currentIndex = 0;
    }

    public void start() {
        System.out.println("Welcome to Simon Clone!");
        System.out.println("Repeat the sequence by entering the numbers separated by spaces.");
        System.out.println("Get ready...");

        try {
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        while (true) {
            System.out.println("Level " + level);
            generateSequence(level);

            boolean success = playSequence();
            if (!success) {
                System.out.println("Game Over!");
                System.out.println("You reached level " + level);
                break;
            }

            level++;
            currentIndex = 0;
            System.out.println("Congratulations! Next level...");
            try {
                Thread.sleep(1500);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }

    private void generateSequence(int length) {
        Random random = new Random();
        for (int i = 0; i < length; i++) {
            sequence.add(random.nextInt(4) + 1);
        }
    }

    private boolean playSequence() {
        for (int num : sequence) {
            System.out.print(num + " ");
            try {
                Thread.sleep(1500);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
        System.out.println();

        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        String[] inputArray = input.split(" ");

        for (String numStr : inputArray) {
            int number = Integer.parseInt(numStr);
            if (number != sequence.get(currentIndex)) {
                return false;
            }
            currentIndex++;
        }

        return true;
    }
}

public class Main {
    public static void main(String[] args) {
        Game game = new Game();
        game.start();
    }
}

File: code_challenge_solution/c#/177 Simon.cs
```csharp
using System;
using System.Collections.Generic;

namespace SimonClone
{
    class Program
    {
        static List<int> sequence = new List<int>();
        static Random random = new Random();
        static int score = 0;
        static bool gameOver = false;

        static void Main(string[] args)
        {
            Console.WriteLine("Welcome to Simon Clone!");
            Console.WriteLine("Repeat the sequence by pressing the corresponding keys.");
            Console.WriteLine("Press any key to start...");
            Console.ReadKey();
            Console.Clear();

            while (!gameOver)
            {
                sequence.Add(random.Next(0, 4));
                PlaySequence();
                GetUserInput();
                score++;
            }

            Console.WriteLine($"Game Over! Your score is: {score}");
            Console.WriteLine("Press any key to exit...");
            Console.ReadKey();
        }

        static void PlaySequence()
        {
            foreach (int color in sequence)
            {
                Console.Clear();
                Console.WriteLine("Simon Clone");
                Console.WriteLine("--------------");

                if (color == 0)
                {
                    Console.WriteLine("Red");
                    Console.Beep(200, 500);
                }
                else if (color == 1)
                {
                    Console.WriteLine("Green");
                    Console.Beep(300, 500);
                }
                else if (color == 2)
                {
                    Console.WriteLine("Blue");
                    Console.Beep(400, 500);
                }
                else if (color == 3)
                {
                    Console.WriteLine("Yellow");
                    Console.Beep(500, 500);
                }

                System.Threading.Thread.Sleep(800);
                Console.Clear();
                System.Threading.Thread.Sleep(200);
            }
        }

        static void GetUserInput()
        {
            for (int i = 0; i < sequence.Count; i++)
            {
                Console.WriteLine("Your turn:");
                Console.WriteLine("--------------");
                Console.WriteLine("Red: R");
                Console.WriteLine("Green: G");
                Console.WriteLine("Blue: B");
                Console.WriteLine("Yellow: Y");

                ConsoleKeyInfo keyInfo = Console.ReadKey();
                Console.WriteLine();

                if (keyInfo.Key == ConsoleKey.R && sequence[i] == 0)
                {
                    Console.WriteLine("Correct!");
                    Console.Beep(200, 500);
                }
                else if (keyInfo.Key == ConsoleKey.G && sequence[i] == 1)
                {
                    Console.WriteLine("Correct!");
                    Console.Beep(300, 500);
                }
                else if (keyInfo.Key == ConsoleKey.B && sequence[i] == 2)
                {
                    Console.WriteLine("Correct!");
                    Console.Beep(400, 500);
                }
                else if (keyInfo.Key == ConsoleKey.Y && sequence[i] == 3)
                {
                    Console.WriteLine("Correct!");
                    Console.Beep(500, 500);
                }
                else
                {
                    Console.WriteLine("Wrong sequence! Game over.");
                    gameOver = true;
                    break;
                }

                System.Threading.Thread.Sleep(800);
                Console.Clear();
                System.Threading.Thread.Sleep(200);
            }
        }
    }
}
```